actors/abi/sector.go

```diff
@@ -64,140 +64,166 @@ func NewStoragePower(n int64) StoragePower {
+	case RegisteredSealProof_StackedDrg32GiBV1:
 		return 2349, nil
-	case RegisteredProof_StackedDRG2KiBSeal:
+	case RegisteredSealProof_StackedDrg2KiBV1:
 		return 2, nil
-	case RegisteredProof_StackedDRG8MiBSeal:
+	case RegisteredSealProof_StackedDrg8MiBV1:
 		return 2, nil
-	case RegisteredProof_StackedDRG512MiBSeal:
+	case RegisteredSealProof_StackedDrg512MiBV1:
 		return 2, nil
 	default:
 		return 0, errors.Errorf("unsupported proof type: %v", p)
 	}
 }
 
+// Returns the partition size, in sectors, associated with a proof type.
+// The partition size is the number of sectors proved in a single PoSt proof.
+func (p RegisteredPoStProof) WindowPoStPartitionSectors() (uint64, error) {
+	// Resolve to seal proof and then compute size from that.
+	sp, err := p.RegisteredSealProof()
+	if err != nil {
+		return 0, err
+	}
+	return sp.WindowPoStPartitionSectors()
+}
+
 // RegisteredWinningPoStProof produces the PoSt-specific RegisteredProof corresponding
 // to the receiving RegisteredProof.
-func (p RegisteredProof) RegisteredWinningPoStProof() (RegisteredProof, error) {
+func (p RegisteredSealProof) RegisteredWinningPoStProof() (RegisteredPoStProof, error) {
 	switch p {
-	case RegisteredProof_StackedDRG64GiBSeal, RegisteredProof_StackedDRG64GiBWindowPoSt, RegisteredProof_StackedDRG64GiBWinningPoSt:
-		return RegisteredProof_StackedDRG64GiBWinningPoSt, nil
-	case RegisteredProof_StackedDRG32GiBSeal, RegisteredProof_StackedDRG32GiBWindowPoSt, RegisteredProof_StackedDRG32GiBWinningPoSt:
-		return RegisteredProof_StackedDRG32GiBWinningPoSt, nil
-	case RegisteredProof_StackedDRG2KiBSeal, RegisteredProof_StackedDRG2KiBWindowPoSt, RegisteredProof_StackedDRG2KiBWinningPoSt:
-		return RegisteredProof_StackedDRG2KiBWinningPoSt, nil
-	case RegisteredProof_StackedDRG8MiBSeal, RegisteredProof_StackedDRG8MiBWindowPoSt, RegisteredProof_StackedDRG8MiBWinningPoSt:
-		return RegisteredProof_StackedDRG8MiBWinningPoSt, nil
-	case RegisteredProof_StackedDRG512MiBSeal, RegisteredProof_StackedDRG512MiBWindowPoSt, RegisteredProof_StackedDRG512MiBWinningPoSt:
-		return RegisteredProof_StackedDRG512MiBWinningPoSt, nil
+	case RegisteredSealProof_StackedDrg64GiBV1:
+		return RegisteredPoStProof_StackedDrgWinning64GiBV1, nil
+	case RegisteredSealProof_StackedDrg32GiBV1:
+		return RegisteredPoStProof_StackedDrgWinning32GiBV1, nil
+	case RegisteredSealProof_StackedDrg2KiBV1:
+		return RegisteredPoStProof_StackedDrgWinning2KiBV1, nil
+	case RegisteredSealProof_StackedDrg8MiBV1:
+		return RegisteredPoStProof_StackedDrgWinning8MiBV1, nil
+	case RegisteredSealProof_StackedDrg512MiBV1:
+		return RegisteredPoStProof_StackedDrgWinning512MiBV1, nil
 	default:
 		return 0, errors.Errorf("unsupported mapping from %+v to PoSt-specific RegisteredProof", p)
 	}
 }
 
+// RegisteredWinningPoStProof produces the PoSt-specific RegisteredProof corresponding
+// to the receiving RegisteredProof.
+func (p RegisteredPoStProof) RegisteredWinningPoStProof() (RegisteredPoStProof, error) {
```

Comment:
We probably don't need this method, providing PoStProof->PoStProof. It's only a convenience and I can't think of an immediate need given the SealProof->WinningPoStProof method.

(The others are also now only convenience, but we understand a use case for them all so it makes sense to provide it).

Reply:
Done.